packages/@aws-cdk/aws-codecommit/lib/repository.ts

```diff
@@ -182,6 +186,24 @@ class ImportedRepositoryRef extends RepositoryRef {
         });
         this.repositoryName = props.repositoryName;
     }
+
+    public get repositoryCloneUrlHttp() {
+        return new cdk.FnConcat('https://git-codecommit.',
+                                new cdk.AwsRegion(),
+                                '.',
+                                new cdk.AwsURLSuffix(),
+                                '/v1/repos/',
+                                this.repositoryName).toString();
+    }
+
+    public get repositoryCloneUrlSsh() {
+        return new cdk.FnConcat('ssh://git-codecommit.',
+                                new cdk.AwsRegion(),
+                                '.',
+                                new cdk.AwsURLSuffix(),
+                                '/v1/repos/',
+                                this.repositoryName).toString();
+    }
```

Comment:
There seems to be a fair amount of duplication between the 2 methods - does extracting a third, private one, make sense here?

Reply:
That's fair, let me do that...